gtsam/sfm/TranslationRecovery.h

```diff
@@ -131,5 +165,25 @@ class TranslationRecovery {
    */
   static TranslationEdges SimulateMeasurements(
       const Values &poses, const std::vector<KeyPair> &edges);
+
+ private:
+  /**
+   * @brief Gets the key of the variable being optimized among multiple input
+   * variables that have the same translation.
+   *
+   * @param i key of input variable.
+   * @return Key of optimized variable - same as input if it does not have any
+   * zero-translation edges.
+   */
+  Key getUniqueKey(const Key i) const;
```

Comment:
I have no idea what this does. we can chat about it.

Reply:
If there are multiple input nodes that share the same translation, i.e have a zero unit translation edge between them, only one of them will be added to the factor graph. This function will return the value of the Key that was added to the graph for an input node. If there is no zero-translation edge, the output is the same as input. 

It is used each time we add the betweenFactors or any other priors, we need to find that "root" node that was added to the graph. I renamed the method, hopefully its more clear.